cadence/mapping/grpc/domain.py

```diff
@@ -31,16 +34,112 @@ def proto_list_domains_response_to_dataclass(
     ) if list_domains else None
 
 
-def describe_domain_request_dataclass_to_proto(describe_domain_request: DescribeDomainRequest) -> service_domain_pb2.DescribeDomainRequest:
-    a = service_domain_pb2.DescribeDomainRequest(
+def describe_domain_request_dataclass_to_proto(
+        describe_domain_request: DescribeDomainRequest) -> service_domain_pb2.DescribeDomainRequest:
+    return service_domain_pb2.DescribeDomainRequest(
         id=describe_domain_request.uuid,
         name=describe_domain_request.name,
     )
-    return a
 
 
-def proto_describe_domain_response_to_describe_domain_response_dataclass(describe_domain_response: service_domain_pb2.DescribeDomainResponse) -> DescribeDomainResponse:
-    return proto_domain_to_describe_domain_response_dataclass(describe_domain_response.domain) if describe_domain_response else None
+DomainUpdateDescriptionField = "description"
+DomainUpdateOwnerEmailField = "owner_email"
+DomainUpdateDataField = "data"
+DomainUpdateRetentionPeriodField = "workflow_execution_retention_period"
+DomainUpdateBadBinariesField = "bad_binaries"
+DomainUpdateHistoryArchivalStatusField = "history_archival_status"
+DomainUpdateHistoryArchivalURIField = "history_archival_uri"
+DomainUpdateVisibilityArchivalStatusField = "visibility_archival_status"
+DomainUpdateVisibilityArchivalURIField = "visibility_archival_uri"
+DomainUpdateActiveClusterNameField = "active_cluster_name"
+DomainUpdateClustersField = "clusters"
+DomainUpdateDeleteBadBinaryField = "delete_bad_binary"
+DomainUpdateFailoverTimeoutField = "failover_timeout"
+
+
+def update_domain_request_dataclass_to_proto(
```

Comment:
It seems too tedious...
Can we use refection?  It seems possible by using `get_attribute` and `set_attribute`. 

I am okay to let you make the choice. But I belive it will save you lots of code/time when implementing other APIs. 

Note:
When doing that, the gRPC structure is flattening the fields, we could also do similar to data class one so that it's easier to map. 
Also the type converting needs to be smarter based on source and target types. 
We

Reply:
Not really sure about the concept of refection so I'll google it. The function became a copy of the [go-client part](https://github.com/uber/cadence/blob/064378890cd0186b50edd36dd679b230c44e45fd/common/types/mapper/proto/api.go#L4009) and the only reason is lengthy is because `update_mask` has to have the fields to be updated. 

Let me see if refection might make things easier because I 100% agree with you